File: ios/MapsHereConfigModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(MapsHereConfigModule, NSObject)

RCT_EXTERN_METHOD(initializeHereSDK:(NSString *)accessKeyID
                  withAccessKeySecret:(NSString *)accessKeySecret
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/MapsHereConfigModule.swift
import heresdk

@objc(MapsHereConfigModule)
class MapsHereConfigModule: NSObject {
    
    @objc(initializeHereSDK:withAccessKeySecret:withResolver:withRejecter:)
    func initializeHereSDK(
        accessKeyID: String,
        accessKeySecret: String,
        resolve: RCTPromiseResolveBlock,
        reject: RCTPromiseRejectBlock) -> Void {
            
            let options = SDKOptions(
                accessKeyId: accessKeyID,
                accessKeySecret: accessKeySecret)
            
            do {
                try SDKNativeEngine.makeSharedInstance(options: options)
                print("SDKOptions engine started")
                resolve("Here SDK initialized")
            } catch let error {
                reject(error)
            }
        }
}

